travis.pl

```diff
@@ -0,0 +1,408 @@
+               "sudo wget -q https://xcat.org/files/xcat/xcat-core/devel/Ubuntu/core-snap/core-debs-snap.tar.bz2",
+               "sudo tar -jxvf core-debs-snap.tar.bz2",
+               "cd xcat-core && sudo ./mklocalrepo.sh",
+               "sudo chmod 777 /etc/apt/sources.list",
+               "sudo echo \"deb [arch=amd64 allow-insecure=yes] http://xcat.org/files/xcat/repos/apt/devel/xcat-dep xenial main\" >> /etc/apt/sources.list",
+               "sudo echo \"deb [arch=ppc64el allow-insecure=yes] http://xcat.org/files/xcat/repos/apt/devel/xcat-dep xenial main\" >> /etc/apt/sources.list",
+               "sudo wget -q -O - \"http://xcat.org/files/xcat/repos/apt/apt.key\" | sudo apt-key add -",
+               "sudo apt-get -qq update");
+    my @output;
+    foreach my $cmd (@cmds){
+        print "[install_xcat] running $cmd\n";
+        @output = runcmd("$cmd");
+        if ($::RUNCMD_RC){
+            print RED "[install_xcat] $cmd. ...[Failed]\n";
+            print "[install_xcat] error message:\n";
+            #print Dumper \@output;
+            #$check_result_str .= "> **INSTALL XCAT ERROR** : Please click ``Details`` label in ``Merge pull request`` box for detailed information ";
+            #send_back_comment("$check_result_str");
+            return 1;
+        }
+    }
+
+    my $cmd = "sudo apt-get install xcat --allow-remove-essential --allow-unauthenticated";
+    @output = runcmd("$cmd");
+    if($::RUNCMD_RC){
+        my $lastline = $output[-1];
+        $lastline =~ s/[\r\n\t\\"']*//g;
+        print "[install_xcat] $cmd ....[Failed]\n";
+        print ">>>>>Dumper the output of '$cmd'\n";
+        #print Dumper \@output;
+        #$check_result_str .= "> **INSTALL XCAT ERROR** : Please click ``Details`` label in ``Merge pull request`` box for detailed information";
+        #send_back_comment("$check_result_str");
+        return 1;
+    }else{
+        print "[install_xcat] $cmd ....[Pass]\n";
+    }
+
+    my $sshcmd = "sudo bash -c 'echo StrictHostKeyChecking no >> ~root/.ssh/config && echo UserKnownHostsFile /dev/null >> ~root/.ssh/config && cat ~root/.ssh/id_rsa.pub >> ~root/.ssh/authorized_keys'";
+    @output = runcmd("$sshcmd");
+    if ($::RUNCMD_RC){
+        print RED "[install_xcat] $sshcmd. ...[Failed]\n";
+        print Dumper \@output;
+        return 1;
+    }
+
+    return 0;
+}
+
+sub install_xcattest {
+    my $cmd = "sudo apt-get install xcat-test --allow-remove-essential --allow-unauthenticated";
+    my @output = runcmd("$cmd");
+    if($::RUNCMD_RC){
+         print RED "[install_xcattest] $cmd ....[Failed]\n";
+         print Dumper \@output;
+         return 1;
+    }else{
+        print "[install_xcattest] $cmd .....:\n";
+        #print Dumper \@output;
+    }
+
+    # get newest xcat-inventory cases from github and update
+    my $gitdir = "/tmp/xcat-core";
+    mkdir($gitdir);
+    $cmd = "sudo git clone -b case_inv https://github.com/xuweibj/xcat-core.git $gitdir";
```

Comment:
what is this for?

Reply:
Modified to xcat2 repo and master branch.